### PR TITLE
445 replace oraclekey enums used in treasury buyout extension

### LINKED ISSUE #445: Replace OracleKey enums used in treasury-buyout-extension
### Context

At this moment, `treasury-buyout-extension` is using `oracle` pallet from Spacewalk to fetch asset prices. Since `CustomMetadata` struct containing DIA keys was added to our runtimes, we can use `dia-oracle` pallet directly for this use case.

### Requirement

Modify `PriceGetter` implementation for all runtimes so that it uses DIA fields from `CustomMetadata` struct instead of `OracleKey` [enum](https://github.com/pendulum-chain/pendulum/blob/dd8eb3d55ac2933bfc375a13b7a9d2d73ca4ea86/runtime/pendulum/src/lib.rs#L1409-L1410).

Note: DIA fields should be checked if they are empty.

### PR DESCRIPTION
**Pull request by bogdanS98**
This PR modifies `PriceGetter` trait implementation by replacing usage of `oracle` pallet with `dia-oracle`. 
`PriceGetter` is a trait that is used to fetch prices of assets when doing buyouts.

Closes #445.

### CODE REVIEW

**Review by ebma — COMMENTED** [review]
Let's generify and deduplicate the `OraclePriceGetter` and put it into the `common` directory, similar to what I did with the struct [here](https://github.com/pendulum-chain/pendulum/blob/1c079bd7af4141bf7f86675b32f4168a6fa7939e/runtime/common/src/asset_registry.rs#L81).

File: runtime/common/src/lib.rs [hunk 1]
```diff
@@ -184,3 +191,50 @@ impl<AssetRegistry: Inspect<AssetId = CurrencyId,Balance = Balance, CustomMetada
 			.ok_or(location)
 	}
 }
+
+
+pub struct OraclePriceGetter<Runtime>(PhantomData<Runtime>);
+impl < Runtime: treasury_buyout_extension::Config + dia_oracle::Config + orml_asset_registry::Config<AssetId = CurrencyId, CustomMetadata = CustomMetadata> > PriceGetter<CurrencyId> for OraclePriceGetter<Runtime> 
+{
```
Comment (ebma) [inline runtime/common/src/lib.rs:198]: The formatting here looks a bit off. Does the code compile?

Comment (bogdanS98) [reply runtime/common/src/lib.rs:198]: Yes, it compiles. I added another commit with formatting 👍🏼

**Review by ebma — APPROVED** [review]
LGTM!

**Pull request merged**